makefile

```diff
@@ -278,12 +278,12 @@ endif
 ##
 .PHONY: clean clean-deps clean-all
 
-clean:
-	$(RM) -r test
-	$(RM) $(wildcard $(patsubst %.stan,%.d,$(TEST_MODELS)))
-	$(RM) $(wildcard $(patsubst %.stan,%.hpp,$(TEST_MODELS)))
-	$(RM) $(wildcard $(patsubst %.stan,%.o,$(TEST_MODELS)))
-	$(RM) $(wildcard $(patsubst %.stan,%$(EXE),$(TEST_MODELS)))
+clean: clean-tests
+	$(RM) -r bin
```

Comment:
I don't think we want to delete the entire directory since the release tarballs will have the `linux-stanc`, `windows-stanc`, etc files right?